fintorch/datasets/ellipticpp.py

```diff
@@ -351,3 +359,173 @@ def process(self):
+        neg_sampling_ratio: float = 2.0,
+        num_neighbors: List[int] = [10, 30],
+        batch_size: int = 128,
+        neg_sampling: str = "binary",
+        num_workers: int = -1,
+        force_reload=False,
+    ) -> None:
+        super().__init__()
+
+        assert (isinstance(edge, tuple)
+                and len(edge) == 3), "edge must be a tuple of length 3"
+        assert (isinstance(num_val, float) and
+                0 <= num_val <= 1), "num_val must be a float between 0 and 1"
+        assert (isinstance(num_test, float) and
+                0 <= num_test <= 1), "num_test must be a float between 0 and 1"
+        assert (isinstance(disjoint_train_ratio, float)
+                and 0 <= disjoint_train_ratio <= 1
+                ), "disjoint_train_ratio must be a float between 0 and 1"
+        assert (isinstance(neg_sampling_ratio, float) and neg_sampling_ratio
+                > 0), "neg_sampling_ratio must be a positive float"
+        assert isinstance(num_neighbors, list) and all(
+            isinstance(n, int)
+            for n in num_neighbors), "num_neighbors must be a list of integers"
+        assert (isinstance(batch_size, int)
+                and batch_size > 0), "batch_size must be a positive integer"
+        assert isinstance(neg_sampling, str), "neg_sampling must be a string"
+
+        self.edge = edge
+        self.num_val = num_val
+        self.num_test = num_test
+        self.disjoint_train_ratio = disjoint_train_ratio
+        self.neg_sampling_ratio = neg_sampling_ratio
+        self.num_neighbors = num_neighbors
+        self.batch_size = batch_size
+        self.neg_sampling = neg_sampling
+        self.force_reload = force_reload
+
+        if num_workers == -1:
+            self.num_workers = multiprocessing.cpu_count() - 1
+        else:
+            self.num_workers = num_workers
+
+    def setup(self, stage=None):
+        dataset = TransactionActorDataset("~/.fintorch_data",
+                                          force_reload=self.force_reload)
+        self.dataset = dataset[0]
+        self.split_dataset(dataset[0])
+
+    def split_dataset(self, dataset):
+        """
+        Splits the given dataset into training, validation, and test sets and makes them available
+        as self.train_data, self.val_data, and self.test_data
+
+        Args:
+            dataset: The input dataset to be split.
+
+        Returns:
+            None
+        """
+
+        transform = T.RandomLinkSplit(
+            num_val=0.1,
+            num_test=0.1,
+            disjoint_train_ratio=0.3,
+            neg_sampling_ratio=2.0,
+            edge_types=[
+                ("wallets", "to", "transactions"),
+                ("transactions", "to", "wallets"),
+                ("wallets", "to", "wallets"),
+                ("transactions", "to", "transactions"),
+            ],
+        )
+        self.train_data, self.val_data, self.test_data = transform(dataset)
```

Comment:
Transform per time slice and combine into train, validate, and test.

Reply:
We have a set of graphs for each time step. Therefore, similar to https://github.com/pyg-team/pytorch_geometric/pull/7211 we might be able to solve it.